holidays/countries/sao_tome_and_principe.py

```diff
@@ -0,0 +1,109 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.groups import ChristianHolidays, InternationalHolidays
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON, SAT_TO_PREV_FRI
+
+
+class SaoTomeAndPrincipe(ObservedHolidayBase, ChristianHolidays, InternationalHolidays):
+    """São Tomé and Príncipe holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_São_Tomé_and_Príncipe>
+        * <https://www.timeanddate.com/holidays/sao-tome-and-principe/>
+        * <https://www.saotomeexpert.pt/en/sao-tome-public-holidays/>
+    """
+
+    country = "ST"
+    default_language = "pt_ST"
+    # %s (observed).
+    observed_label = tr("%s (observado)")
+    supported_languages = ("en_US", "pt_ST")
+    start_year = 2014
+    subdivisions = (
+        "01",
+        "02",
+        "03",
+        "04",
+        "05",
+        "06",
+        "P",
+    )
+    subdivisions_aliases = {
+        # Districts.
+        "Água Grande": "01",
+        "Cantagalo": "02",
+        "Caué": "03",
+        "Lembá": "04",
+        "Lobata": "05",
+        "Mé-Zóchi": "06",
+        # Autonomous Region.
+        "Príncipe": "P",
+    }
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        kwargs.setdefault("observed_rule", SUN_TO_NEXT_MON + SAT_TO_PREV_FRI)
+        # Holidays are observed on the next Monday if on Sunday, or previous Friday if on Saturday.
+        # Based on common government practices since at least 2020.
+        kwargs.setdefault("observed_since", 2020)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_observed(self._add_new_years_day(tr("Ano Novo")))
```

Comment:
This is for 2022's case which is instead observed on `2021-12-31`:
```suggestion
        # New Year's Day.
        name = tr("Ano Novo")
        self._add_observed(self._add_new_years_day(name))
        self._add_observed(self._next_year_new_years_day, name=name, rule=SAT_TO_PREV_FRI)
```